Sources/Types.swift

```diff
@@ -34,15 +34,73 @@ public typealias GetState = () -> State
 /// Typealias for the `Store.anyDispatch` function with the ability of managing the output with a promise
 public typealias AnyDispatch = (_: Dispatchable) -> Promise<Any>
 
-/// Entity capable of executing task asynchronously. This can be useful in tests to control asynchronous tasks
-public protocol AsyncProvider {
-  /// Calling this method will enqueue a closure to be executed asynchronously
+/// See `Executor`
+@available(*, deprecated, message: "Use the new `Executor` instead")
+public typealias AsyncProvider = Executor
+
+/// Entity capable of executing a closure. This can be useful in tests to control where and when tasks
+/// are executed.
+///
+/// - warning: Katana assumes that implementations of this protocol will always guarantee that the tasks
+/// given as arguments are never executed in parallel.
+public protocol Executor {
+  /// Calling this method will enqueue a task to be executed asynchronously
+  @available(*, deprecated, message: "Use the new `executeAsync` instead")
   func execute(_ closure: @escaping () -> Void)
```

Comment:
This is deprecated but when I create my custom Executor I am still forced to define it, right?

Reply:
Yep, it's deprecated from the caller's perspective :sisi:

Reply:
It's a little bit ugly that I have to keep defining old deprecated methods, but fine for me if we want to have a non-breaking update that fixes the issue.

Reply:
That was the idea. When we'll release a new major we will remove/rename this method and all should be good again